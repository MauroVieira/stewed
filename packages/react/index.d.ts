/**
 * Required to avoid errors when importing these types of files
 */

declare module "*.svg?url" {
  const content: string;
  export default content;
}

declare module "*.svg" {
  const content: React.FunctionComponent<React.SVGAttributes<SVGElement>>;
  export default content;
}

declare module "*.md" {
  const content: string;
  export default content;
}

declare module "*.scss" {
  const classes: { [key: string]: string };
  export default classes;
}
