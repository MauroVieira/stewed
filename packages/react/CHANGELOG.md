# @stewed/react

## 0.1.0

### Minor Changes

- a302a43: Update all packages

### Patch Changes

- Updated dependencies [a302a43]
  - @stewed/utilities@0.2.0
  - @stewed/hooks@0.1.0
  - @stewed/tokens@0.2.0
