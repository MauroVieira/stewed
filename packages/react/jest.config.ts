import { jestConfig } from "@harmonix-hub/testing";

module.exports = {
  ...jestConfig(),
  testEnvironment: "jsdom",
  moduleNameMapper: {
    "\\.(css|less|scss|sss|styl)$": "<rootDir>/node_modules/jest-css-modules",
  }
};
