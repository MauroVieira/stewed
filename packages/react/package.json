{
  "name": "@stewed/react",
  "version": "0.0.0",
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "sideEffects": false,
  "license": "MIT",
  "files": [
    "dist/**"
  ],
  "scripts": {
    "dev": "npx vite",
    "build": "npx vite build",
    "lint": "TIMING=1 eslint \"src/**/*.ts*\"",
    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist"
  },
  "devDependencies": {
    "@stewed/tsconfig": "workspace:*",
    "@stewed/utils": "workspace:*",
    "@stewed/lint": "workspace:*",
    "@types/react": "^18.0.9",
    "@types/react-dom": "^18.0.4",
    "eslint": "^8.15.0",
    "react": "^18.1.0",
    "tsup": "^5.10.1",
    "typescript": "^4.5.3",
    "sass": "^1.58.0",
    "@vitejs/plugin-react": "^1.3.2",
    "vite": "^2.9.15"
  },
  "publishConfig": {
    "access": "public"
  }
}
