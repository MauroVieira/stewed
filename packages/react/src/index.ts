export * from "./accordion";
export * from "./alert";
export * from "./aspect-ratio";
export * from "./avatar";
export * from "./badge";
export * from "./box";
export * from "./button";
export * from "./card";
export * from "./carousel";
export * from "./checkbox";
export * from "./divider";
export * from "./list-box";
export * from "./progress";
export * from "./radio";
export * from "./tabs";
export * from "./tag";
export * from "./text-area";
export * from "./text-field";
export * from "./theme";
export * from "./theme";
export * from "./toggle";
export * from "./typography";

// Utilities
export * from "./types";
