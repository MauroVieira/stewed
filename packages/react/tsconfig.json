{
  "extends": "@harmonix-hub/tsconfig/react.json",
  "compilerOptions": {
    "outDir": "./dist",
  },
  "include": ["custom.d.ts", "index.ts", "./src/**/*", "./src/index.ts"],
  "exclude": ["dist", "src/**/stories", "src/**/__tests__", "node_modules"]
}
