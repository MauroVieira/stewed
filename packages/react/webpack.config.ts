import path from "path";
// Types
import type { Configuration } from "webpack";

const webpackConfig: Configuration = {
  mode: "production",
  entry: "./src/index.ts",
  output: {
    filename: "index.js",
    library: {
      type: "module",
    },
    path: path.resolve(__dirname, "dist"),
  },
  experiments: {
    outputModule: true,
  },
  optimization: {
    minimize: true,
  },
  module: {
    rules: [
      {
        test: /\.s[ac]ss$/i,
        use: [
          "style-loader",
          {
            loader: "css-loader",
            options: {
              modules: {
                auto: true,
                localIdentName: "[hash:base64]",
              },
              sourceMap: true,
            },
          },
          "sass-loader",
        ],
      },
      {
        test: /\.tsx?$/i,
        exclude: /node_modules/,
        use: {
          loader: "ts-loader",
          options: {
            compilerOptions: {
              sourceMap: true,
            },
          },
        },
      },
    ],
  },
  resolve: {
    extensions: [".tsx", ".ts", ".js"],
  }
};

export default webpackConfig;
