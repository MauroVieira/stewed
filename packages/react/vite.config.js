import { defineConfig } from "vite";
// Node
import { resolve } from "path";
// Plugins
import dts from "vite-plugin-dts";
import peerDepsExternal from "rollup-plugin-peer-deps-external";
import react from "@vitejs/plugin-react";
import { libInjectCss } from "vite-plugin-lib-inject-css";

export default defineConfig({
  plugins: [
    react(),
    libInjectCss(),
    dts({
      copyDtsFiles: true
    }),
  ],
  build: {
    copyPublicDir: false,
    lib: {
      entry: resolve(__dirname, "src/index.ts"),
      formats: ["es"],
    },
    rollupOptions: {
      plugins: [peerDepsExternal()],
      output: {
        assetFileNames: "assets/[name][extname]",
        entryFileNames: "[name].js",
      },
    },
  },
});
