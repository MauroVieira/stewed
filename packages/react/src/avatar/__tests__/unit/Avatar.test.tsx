import React from "react";
// UI Components
import { Avatar } from "../../index";
// Utilities
import { render } from "@testing-library/react";

describe("Avatar", () => {
  describe("Snapshots", () => {
    it("should renders default component", () => {
      const { asFragment } = render(<Avatar name="Emma Clark" />);

      expect(asFragment()).toMatchSnapshot();
    });

    it("should renders component as anchor", () => {
      const { asFragment } = render(<Avatar as="a" href="#" name="Emma Clark" />);

      expect(asFragment()).toMatchSnapshot();
    });

    it("should render image", async () => {
      const img = {
        src: "path/to/image.jpg",
        alt: "Image",
      };

      const { asFragment } = render(<Avatar name="Emma Clark" image={{ src: img.src }} />);

      expect(asFragment()).toMatchSnapshot();
    });

    it("should render additional classes", () => {
      const { asFragment } = render(<Avatar className="other-class" name="Emma Clark" />);

      expect(asFragment()).toMatchSnapshot();
    });

    it("should applies correct skin classes", () => {
      const { asFragment } = render(<Avatar skin="primary" name="Emma Clark" />);

      expect(asFragment()).toMatchSnapshot();
    });

    it("should applies correct size classes", () => {
      const { asFragment } = render(<Avatar size="lg" name="Emma Clark" />);

      expect(asFragment()).toMatchSnapshot();
    });

    it("should applies square appearance classes", () => {
      const { asFragment } = render(<Avatar appearance="square" name="Emma Clark" />);

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
