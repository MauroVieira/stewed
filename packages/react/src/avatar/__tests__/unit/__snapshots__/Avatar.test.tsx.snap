// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Snapshots should apply 'circle' appearance classes 1`] = `
<div>
  <div
    class="avatar avatar--circle avatar--md avatar--primary"
  >
    EC
  </div>
</div>
`;

exports[`Avatar Snapshots should apply 'critical' skin classes 1`] = `
<div>
  <div
    class="avatar avatar--circle avatar--md avatar--critical"
  >
    EC
  </div>
</div>
`;

exports[`Avatar Snapshots should apply 'lg' size classes 1`] = `
<div>
  <div
    class="avatar avatar--circle avatar--lg avatar--primary"
  >
    EC
  </div>
</div>
`;

exports[`Avatar Snapshots should apply 'md' size classes 1`] = `
<div>
  <div
    class="avatar avatar--circle avatar--md avatar--primary"
  >
    EC
  </div>
</div>
`;

exports[`Avatar Snapshots should apply 'neutral' skin classes 1`] = `
<div>
  <div
    class="avatar avatar--circle avatar--md avatar--neutral"
  >
    EC
  </div>
</div>
`;

exports[`Avatar Snapshots should apply 'primary' skin classes 1`] = `
<div>
  <div
    class="avatar avatar--circle avatar--md avatar--primary"
  >
    EC
  </div>
</div>
`;

exports[`Avatar Snapshots should apply 'secondary' skin classes 1`] = `
<div>
  <div
    class="avatar avatar--circle avatar--md avatar--secondary"
  >
    EC
  </div>
</div>
`;

exports[`Avatar Snapshots should apply 'sm' size classes 1`] = `
<div>
  <div
    class="avatar avatar--circle avatar--sm avatar--primary"
  >
    EC
  </div>
</div>
`;

exports[`Avatar Snapshots should apply 'square' appearance classes 1`] = `
<div>
  <div
    class="avatar avatar--square avatar--md avatar--primary"
  >
    EC
  </div>
</div>
`;

exports[`Avatar Snapshots should apply 'success' skin classes 1`] = `
<div>
  <div
    class="avatar avatar--circle avatar--md avatar--success"
  >
    EC
  </div>
</div>
`;

exports[`Avatar Snapshots should apply 'xl' size classes 1`] = `
<div>
  <div
    class="avatar avatar--circle avatar--xl avatar--primary"
  >
    EC
  </div>
</div>
`;

exports[`Avatar Snapshots should apply 'xs' size classes 1`] = `
<div>
  <div
    class="avatar avatar--circle avatar--xs avatar--primary"
  >
    EC
  </div>
</div>
`;

exports[`Avatar Snapshots should render additional classes 1`] = `
<div>
  <div
    class="avatar avatar--circle avatar--md avatar--primary other-class"
  >
    EC
  </div>
</div>
`;

exports[`Avatar Snapshots should render image 1`] = `
<div>
  <div
    class="avatar avatar--circle avatar--md avatar--primary"
  >
    <span
      class="avatar__loading"
    >
      <div
        class="spinner spinner--white spinner--md"
      >
        <svg
          class="spinner__svg"
          fill="currentColor"
          height="20"
          viewBox="0 0 48 48"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="24"
            cy="4"
            r="4"
          />
          <circle
            cx="12.19"
            cy="7.86"
            r="3.7"
          />
          <circle
            cx="5.02"
            cy="17.68"
            r="3.4"
          />
          <circle
            cx="5.02"
            cy="30.32"
            r="3.1"
          />
          <circle
            cx="12.19"
            cy="40.14"
            r="2.8"
          />
          <circle
            cx="24"
            cy="44"
            r="2.5"
          />
          <circle
            cx="35.81"
            cy="40.14"
            r="2.2"
          />
          <circle
            cx="42.98"
            cy="30.32"
            r="1.9"
          />
          <circle
            cx="42.98"
            cy="17.68"
            r="1.6"
          />
          <circle
            cx="35.81"
            cy="7.86"
            r="1.3"
          />
        </svg>
      </div>
    </span>
    <img
      alt="Emma Clark"
      class="avatar__img"
      src="path/to/image.jpg"
    />
  </div>
</div>
`;

exports[`Avatar Snapshots should renders component as anchor 1`] = `
<div>
  <a
    class="avatar avatar--circle avatar--md avatar--primary"
    href="#"
  >
    EC
  </a>
</div>
`;

exports[`Avatar Snapshots should renders default component 1`] = `
<div>
  <div
    class="avatar avatar--circle avatar--md avatar--primary"
  >
    EC
  </div>
</div>
`;
